### PR TITLE
fix(wallet): batch NFT balance lookups to avoid hitting SimpleHash limit

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/39970

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See STR in issue

### CODE REVIEW

File: components/brave_wallet_ui/common/slices/endpoints/token_balances.endpoints.ts [hunk 1]
```diff
@@ -929,12 +931,29 @@ async function fetchAccountTokenBalanceRegistryForChainId({
 
   const nfts = nonNativeTokens.filter((token) => token.isNft)
 
+  // NFTs
   if (nfts.length) {
-    // NFTs
-    const { balances: nftBalances, errorMessage: nftBalancesErrorMessage } =
-      await jsonRpcService.getNftBalances(
+    // batch nfts into batches of 50 or less,
+    // then fetch balances for each bach
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/common/slices/endpoints/token_balances.endpoints.ts:937]:
```suggestion
    // then fetch balances for each batch
```

**Review by Douglashdaniel — APPROVED** [review]
Approved with a comment!

File: components/brave_wallet_ui/common/slices/endpoints/token_balances.endpoints.ts (hunk 1 above)
Comment (josheleonard) [reply components/brave_wallet_ui/common/slices/endpoints/token_balances.endpoints.ts]: Thanks, fixed the typo

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.70.31 Chromium: 127.0.6533.73 (Official Build) nightly (64-bit)
-- | --
Revision | 72f6d990d010f0da2b56f3ff90fff8de9efca20d
OS | Linux

- Verified steps from issue description
- Verified no console errors when loading account with more than 50plus NFTs

https://github.com/user-attachments/assets/c16ccabb-fa87-48e7-b3f8-4e70ff14d0af

**Pull request merged**